python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -253,6 +301,213 @@ def _process_data(user_pre_parser, user_parser,
+                                  read_file=read_file)
+
+            return_dict = _process_data(user_pre_parser=None,
+                          user_parser=user_parser,
+                          two_phase_feat_ops=[],
+                          in_files=in_files,
+                          num_proc=num_proc,
+                          task_info=f"edge {edge_type}",
+                          ext_mem_workspace=None)
+
+            src_nids = []
+            for _, return_data in return_dict.items():
+                if return_data is None:
+                    continue
+                src_node_ids, _ = return_data
+                src_nids.append(src_node_ids)
+
+            if len(src_nids) > 0:
+                src_nids = np.concatenate(src_nids)
+                src_nids = np.unique(src_nids)
+                if src_ntype not in node_ids:
+                    node_ids[src_ntype] = [src_nids]
+                else:
+                    node_ids[src_ntype].append(src_nids)
+        elif dst_ntype not in node_types:
+            # only dst_ntype do not appear in node_confs
+            logging.info("Destination nodes from <%s> edges do not have node files."
+                         "Will create node id mapping from edges.",
+                         edge_type)
+            user_parser = partial(parse_edge_nid_data,
+                                  parse_src=False,
+                                  parse_dst=True,
+                                  conf=edge_conf,
+                                  read_file=read_file)
+
+            return_dict = _process_data(user_pre_parser=None,
+                          user_parser=user_parser,
+                          two_phase_feat_ops=[],
+                          in_files=in_files,
+                          num_proc=num_proc,
+                          task_info=f"edge {edge_type}",
+                          ext_mem_workspace=None)
+            dst_nids = []
+            for _, return_data in return_dict.items():
+                if return_data is None:
+                    continue
+                _, dst_node_ids = return_data
+                dst_nids.append(dst_node_ids)
+
+            if len(dst_nids) > 0:
+                dst_nids = np.concatenate(dst_nids)
+                dst_nids = np.unique(dst_nids)
+                if dst_ntype not in node_ids:
+                    node_ids[dst_ntype] = [dst_nids]
+                else:
+                    node_ids[dst_ntype].append(dst_nids)
+        else:
+            logging.debug("Both source and destination nodes"
+                          "from <%s> edges have corresponding node files.",
+                          edge_type)
+
+    for ntype, nids in node_ids.items():
+        nids = np.concatenate(nids)
+        nids = np.unique(nids)
+
+        # We don't need to create ID map if the node IDs are integers,
+        # all node Ids are in sequence start from 0 and
+        # the user doesn't force to remap node IDs.
```

Comment:
Will the node_id here still appear in the raw_id_mappings?

Reply:
It will be `NoopMap(len(nids))`.